terraform/cluster/aws/main.tf

```diff
@@ -66,56 +27,35 @@ resource "aws_eks_cluster" "cluster" {
   ]
 }
 
-resource "local_file" "kubeconfig" {
+resource "aws_eks_node_group" "cluster" {
   depends_on = [
-    aws_cloudformation_stack.nodes,
-    aws_iam_role_policy_attachment.cluster_eks_cluster,
-    aws_iam_role_policy_attachment.cluster_eks_service,
-    aws_iam_role_policy_attachment.nodes_ecr,
-    aws_iam_role_policy_attachment.nodes_eks_cni,
-    aws_iam_role_policy_attachment.nodes_eks_worker,
-    aws_route.private-default,
-    aws_route.public-default,
-    aws_route_table.private,
-    aws_route_table.public,
-    aws_route_table_association.private,
-    aws_route_table_association.public,
-    aws_security_group_rule.cluster_egress_control,
-    aws_security_group_rule.cluster_egress_traffic,
-    aws_security_group_rule.cluster_ingress_control,
-    aws_security_group_rule.nodes_egress_internet,
-    aws_security_group_rule.nodes_ingress_control,
-    aws_security_group_rule.nodes_ingress_internal,
-    aws_security_group_rule.nodes_ingress_mtu,
-    aws_security_group_rule.nodes_ingress_traffic,
+    "aws_iam_role_policy_attachment.nodes_ecr",
+    "aws_iam_role_policy_attachment.nodes_eks_cni",
+    "aws_iam_role_policy_attachment.nodes_eks_worker",
   ]
 
-  filename = pathexpand("~/.kube/config.aws.${var.name}")
-  content = templatefile("${path.module}/kubeconfig.tpl", {
-    ca       = aws_eks_cluster.cluster.certificate_authority.0.data
-    cluster  = aws_eks_cluster.cluster.id
-    endpoint = aws_eks_cluster.cluster.endpoint
-  })
-}
+  count = 3
```

Comment:
I couldn't find a reference to the count argument in the docs for this. For the GCP stuff you need to you use initial_count if you enable autoscaling but that didn't seem to be a argument option for this either:  https://www.terraform.io/docs/providers/aws/r/eks_node_group.html . Am I just looking in the wrong place?

Reply:
`count` is a terraform thing. it means create n of these

Reply:
got it. Thanks for clarifying.